File: sample/src/main/java/com/nearit/sample/CustomFirebaseMessagingService.java
package com.nearit.sample;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

import it.near.sdk.push.NearFcmListenerService;


public class CustomFirebaseMessagingService extends FirebaseMessagingService {

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        NearFcmListenerService.processRemoteMessage(remoteMessage);
    }

    @Override
    public void onNewToken(String refreshedToken) {
        NearFcmListenerService.sendRegistrationToServer(refreshedToken);
    }
}

File: sample-kotlin/src/main/java/com/nearit/sample_kotlin/CustomFirebaseMessagingService.kt
package com.nearit.sample_kotlin

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage
import it.near.sdk.push.NearFcmListenerService

class CustomFirebaseMessagingService : FirebaseMessagingService() {

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        NearFcmListenerService.processRemoteMessage(remoteMessage)
    }

}